pkg/cloud_provider/file/file.go

```diff
@@ -215,6 +216,14 @@ var (
 	shareUriRegex    = regexp.MustCompile(`^projects/([^/]+)/locations/([^/]+)/instances/([^/]+)/shares/([^/]+)$`)
 )
 
+// userErrorCodeMap tells how API error types are translated to error codes.
+var userErrorCodeMap = map[int]codes.Code{
```

Comment:
Can you please change isUserError() to use this map too?

Reply:
I think maybe we could change isUserError() to isUserOperationError() as these occur when errors happen when polling the operations (https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/pull/535) . I think we only need isUserOperationError() to include this logic: 

```
	var apiErr *googleapi.Error
	if !errors.As(err, &apiErr) {
		// Fallback to check for expected error code in the error string
		return containsUserErrStr(err)
	}
```

A little background if you were curious... we already have this [logic integrated into waitForOp in the PDCSI Driver](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/bf3736fb82427da66a6126e4deecfdaeee1e300e/pkg/gce-cloud-provider/compute/gce-compute.go#L1127), so that's why we don't need to do this operation logic in PDCSI CodeForError. PDCSI's implementation is nicer, because we could easily convert the op error to a computev1.OperationErrorErrors, but IIRC, Filestore did some weird error wrapping for operations, so the best way to fix this was to string match.

Reply:
I updated the PR to remove any overlap, let me know what you think!